replit_river/message_buffer.py

```diff
@@ -8,13 +8,19 @@
 logger = logging.getLogger(__name__)
 
 
+class MessageBufferClosedError(BaseException):
+    """Raised when a message buffer is closed and is not accepting new messages."""
+
+
 class MessageBuffer:
     """A buffer to store messages and support current updates"""
 
     def __init__(self, max_num_messages: int = MAX_MESSAGE_BUFFER_SIZE):
```

Comment:
now that we have backpressure, it might be beneficial to reduce this a bit to prevent OOMs!

Reply:
Agree, let's shrink to 128?

Reply:
![](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExODl5bG0xcnc1MngzY2cycTZ1eHp4bzgzdTVwbjl2ZHl6bHJleW0xdCZlcD12MV9naWZzX3NlYXJjaCZjdD1n/qDPg6HNz2NfAk/giphy.webp "DO IT!!!")